server/raft_transport.go

```diff
@@ -159,11 +159,13 @@ func (t *rpcTransport) processQueue(nodeID roachpb.NodeID, storeID roachpb.Store
 	done := make(chan *gorpc.Call, cap(ch))
 	var req *storage.RaftMessageRequest
 	protoResp := &storage.RaftMessageResponse{}
+	raftIdleTimer := time.NewTimer(raftIdleTimeout)
 	for {
+		raftIdleTimer.Reset(raftIdleTimeout)
```

Comment:
Good catch; that does seem like an issue. Now that I look closely, there are several other problems with the shutdown-on-idle behavior. A new request could be written to the channel after the decision has been made to shut down the goroutine, and even without any races the goroutine could stop while there are still outstanding requests that will try to write to the `done` channel. 

We have several instances of this pattern where we want a goroutine that goes away when it is idle (here, `gossip.infoStore.runCallbacks`, and my upcoming work on parallelizing raft execution). I'm going to try to refactor the `infoStore` pattern into something reusable.